.github/workflows/notebook.yaml

```diff
@@ -0,0 +1,92 @@
+name: Notebook service CI
+
+on:
+  push:
+    branches: [ "main" ]
+  pull_request:
+    branches: [ "main" ]
+
+jobs:
+  lint-git: 
+    runs-on: ubuntu-latest
+    steps: 
+      - uses: actions/checkout@v3
+        with:
+          # Checkout on HEAD commit of the current pull request HEAD.
+          # As this job is triggered on pull request event, the checkout is by
+          # default on the merge commit.
+          ref: ${{ github.event.pull_request.head.sha }}
+          # By default, `fetch-depth` is set to 1, meaning it would only
+          # consider the last commit. Set to 0 will fetch the entire history
+          # of the branch commits.
+          fetch-depth: 0
```

Comment:
Not sure why we need to checkout the HEAD commit of the current pull request (and consider all commits), instead of the default behavior of checking out the merge commit.
It didn't work when checking out with default values?

Reply:
because it checkouts only on the merge commit and the commit history is not targeted with the default checkout